panel/widgets/chatbox.py

```diff
@@ -0,0 +1,266 @@
+from panel.pane.image import (
+    GIF, JPG, PNG, SVG,
+)
+from panel.viewable import Layoutable
+from panel.widgets.base import CompositeWidget
+from panel.widgets.input import StaticText
+
+
+class MessageRow(CompositeWidget):
+
+    value = param.String(
+        default="",
+        doc="""The message to display""",
+    )
+
+    text_color = param.String(
+        default="white",
+        doc="""Font color of the chat text""",
+    )
+
+    background_color = param.String(
+        default="black",
+        doc="""Background color of the chat bubble""",
+    )
+
+    icon = param.String(default=None, doc="""The icon to display""")
+
+    style = param.Dict(
+        default={},
+        doc="""Dictionary of CSS properties and values""",
+    )
+
+    show_name = param.Boolean(
+        default=True,
+        doc="""Whether to show the name of the user""",
+    )
+
+    _composite_type: ClassVar[Type[ListPanel]] = Column
+
+    def __init__(
+        self,
+        value: AnyStr,
+        text_color: AnyStr,
+        background: AnyStr,
+        icon: AnyStr = None,
+        style: Dict[str, str] = None,
+        **params,
+    ):
+        style = {
+            "color": text_color,
+            "background-color": background,
+            "border-radius": "12px",
+            "padding": "8px",
+        }
+        style.update(params.pop("style", {}))
+        super().__init__(**params)
+
+        # determine alignment
+        message_layout = {
+            p: getattr(self, p)
+            for p in Layoutable.param
+            if p not in ("name", "height", "margin") and getattr(self, p) is not None
+        }
+        # create the message icon
+        icon_params = {
+            "width": 36,
+            "height": 36,
+            "margin": (12, 2, 12, 2),
+            "sizing_mode": "fixed",
+        }
+        if icon is None:
+            icon_style = {
+                "background-color": background,
+                "border-radius": "18px",
+            }
+            self._icon = pn.pane.Markdown(
+                f"##<center>{self.name[0]}</center>",
+                style=icon_style,
+                **icon_params,
+            )
+        elif isinstance(icon, str):
+            pane_type = PNG
+            if icon.endswith(".jpg"):
+                pane_type = JPG
+            elif icon.endswith(".svg"):
+                pane_type = SVG
+            elif icon.endswith(".gif"):
+                pane_type = GIF
+            self._icon = pane_type(icon, **icon_params)
+        else:
+            # maybe a bytes stream?
+            self._icon = panel(icon, **icon_params)
+
+        # create the message bubble
+        box_width = message_layout.get("width", 100)
+        text_width = int(box_width / 2)  # make text start a new line
+        wrapped_text = "\n".join(wrap(value, width=text_width))
+        self._bubble = StaticText(
+            value=wrapped_text,
+            style=style,
```

Comment:
In 1.0 this will be `styles` and may not work as-is, i.e. you might have to provide `stylesheets` instead.

Reply:
Yup! Seems broken. I couldn't simply replace styles -> stylesheets. I also tried to use `stylesheets=[styles]` but I encountered
```
ValueError: List parameter 'stylesheets' must be a list, not an object of type <class 'dict'>.
```

And then without wrapping a [] around it:
```
ValueError: failed to validate Div(id='p1130', ...).stylesheets: expected an element of List(Either(Instance(StyleSheet), String, Dict(String, Either(Dict(String, Nullable(String)), Instance(Styles))))), got seq with invalid items [{'color': 'grey'}]
```
 `List(Dict(String, Either(Dict(String, Nullable(String)))` should work

<img width="978" alt="image" src="https://user-images.githubusercontent.com/15331990/235070889-776c8bca-d1c8-4b56-b1e4-4382d9380b1c.png">

Reply:
Try to take a look at this:  https://pyviz-dev.github.io/panel/upgrade.html#component-stylesheets

Reply:
Thanks, that was helpful! I think using `styles` is more suitable in this case.